### PR TITLE
Move runtime symbols out of main intermediate package

### PR DESCRIPTION
**Pull request by NikolaMilosavljevic**
This is a follow-up on https://github.com/dotnet/arcade/pull/14015

Size of `runtime` repo's intermediate packages have grown significantly in 9.0 branches, due to some expected native symbols work. This is tracked with https://github.com/dotnet/source-build/issues/3627 and https://github.com/dotnet/runtime/issues/92240

We are currently adding symbols archive to main intermediate package, which is now over 400MB. With symbols archive, it would be over 500MB AzDO limit.

This PR enables repos to specify which intermediate package should carry the symbols archive, by setting `SymbolsIntermediateNupkgCategory` property. By default, archive will still be added to the main intermediate package.

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildIntermediate.proj [hunk 1]
```diff
@@ -80,8 +80,12 @@
     </ItemGroup>
   </Target>
 
-  <!-- Create symbols archive and include it in the "main" intermediate nupkg. -->
-  <Target Name="GetSymbolsArchive" Condition="'$(SupplementalIntermediateNupkgCategory)' == ''">
+  <!--
+    Create symbols archive and include it in the "main" intermediate nupkg,
+    except for 'runtime' repo where "runtime" package is smaller and more appropriate.
+  -->
+  <Target Name="GetSymbolsArchive"
```
Comment (MichaelSimons) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildIntermediate.proj:87]: Is it possible to make this more abstract within arcade and configurable at the repo level?  For example, have arcade define an overridable property that allows the repo to specify the supplemental package to add the symbols to.   It is possible other repos could encounter this size limit and want control where the symbols get packaged.

Comment (NikolaMilosavljevic) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildIntermediate.proj:87]: Sure, that's much better long-term. I'll update the PR and create one for `runtime` repo.

Comment (NikolaMilosavljevic) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildIntermediate.proj:87]: How about a new property, i.e. `IntermediateNupkgCategoryForSymbolsArchive`, that repos can define in their SourceBuild.props? If its defined we'd use it to determine to which package to add the symbols archive.

Comment (NikolaMilosavljevic) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildIntermediate.proj:89]: Fixed with https://github.com/dotnet/arcade/pull/14040/commits/296923b42fed92596aeed1239409fc5265f9f11a

Preparing `runtime` PR - will link it as soon as it's available.

Comment (NikolaMilosavljevic) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/SourceBuild/SourceBuildIntermediate.proj:89]: `runtime` PR: https://github.com/dotnet/runtime/pull/92428

**Review by MichaelSimons — APPROVED** [review]

**Comment by NikolaMilosavljevic** [comment]
@dotnet/dnceng - test failure is unrelated to my PR - also failing in other open `arcade` PRs like https://github.com/dotnet/arcade/pull/14049

Can this be merged? We need this fix to unblock https://github.com/dotnet/runtime/issues/92240

**Comment by NikolaMilosavljevic** [comment]
@mmitche @missymessa @riarenas - can you help me merge this on red? Test failure is unrelated to my PR - also failing in other open arcade PRs like https://github.com/dotnet/arcade/pull/14049

We need this fix to unblock https://github.com/dotnet/runtime/issues/92240

**Comment by missymessa** [comment]
@NikolaMilosavljevic we're working on either getting Known Issues configured for these failing tests in Arcade so we can merge-on-red, or will disable the tests until they can get sorted. They're blocking a lot of PRs.

**Comment by markwilkie** [comment]
The plan according to @missymessa is to disable the tests for now.

**Comment by MichaelSimons** [comment]
> > The plan according to @missymessa is to disable the tests for now.
> 
> What is the timeline for this? This PR is blocking 8.0 work.

@NikolaMilosavljevic kindly corrected me in that this is not 8.0 work.  I crossed my threads.  Apologies.  Still would be nice to get PRs unblocked.

**Pull request merged**